hw/ip/entropy_src/doc/_index.md

```diff
@@ -0,0 +1,210 @@
+The `Entropy ready` bit in the {{< regref "ESSTATUS" >}} register will signal to the software when noise bits should read from the {{< regref "ESNOISE" >}} register.
+While the software is reading the FIFO, additional noise bits are being captured by the FIFO as long as the FIFO is not full.
+The software will do 32-bit register reads of the {{< regref "ESNOISE" >}} register to retrieve the noise bits.
+Each read will automatically pop an entry off the head of the FIFO.
+New noise bits can enter the FIFO at the tail concurrently.
+
+Since the noise flow is continuous, these is no tracking state machine required for the design.
+Instead, the threshold register is used to set the point of notification to the software.
+Notification can be either by polling or by an interrupt.
+For debug, a FIFO depth register can be read to find out the current state of the FIFO at any time. 
+
+An additional feature is the presence of a noise rate register.
+The purpose of this register is to simulation slower noise generation sources.
+This will aid firmware design and debug when trying to build handle all noise source types and rates. 
+
+### Interrupts
+
+The ENTROPY_SRC module has two interrupts: `es_noise_valid` and `es_noise_fifo_err`.
+
+The `es_noise_valid` interrupt should be used when a noise source has been implemented that is relatively slow.
+For this revision, the noise source is very fast, and polling the es_status bit should be done.
+
+The `es_noise_fifo_err` interrupt will fire when the noise FIFO has a malfunction.
+The conditions that cause this to happen are either when there is a push to a full FIFO or a pull from an empty FIFO.
+
+
+### Noise FIFO Operational Sequence
+
+{{< wavejson >}}
+{signal: [
+   {name: 'clk'                 , wave: 'p.......|............'},
+   {name: 'es_enable'           , wave: '01......|............'},
+   {name: 'es_init'             , wave: '010.....|............'},
+   {name: 'fifo_push_capt_noise', wave: '0..1...0|.10.10.10.10', data: ['ph0','ph1','ph2','ph3']},
+   {name: 'noise_bits'          , wave: 'x..3456x|.3x.4x.5x.6x', data: ['es0','es1','es2','es3','es4','es5','es6','es7']},
+   {name: 'fifo_full'           , wave: '0......1|.01.01.01.01'},
+   {name: 'fifo_empty'          , wave: '1..0....|............'},
+   {name: 'fifo_pop_read_noise' , wave: '0.......|10.10.10.10.'},
+   {name: 'threshold reg'       , wave: '3.......|............', data: ['4']},
+  {},
+]}
+{{< /wavejson >}}
+
+
+# Programmers Guide
+
+## Initialization
+
+The following code snippet demonstrates initializing the ENTROPY_SRC block for noise bit generation.
+
+```cpp
+
+void entropy_src_init(unsigned int seed) {
+
+  // set the seed value
+  *ENTROPY_SRC_ES_SEED_REG = seed;
+
+  // set the configuration enable bit
+  *ENTROPY_SRC_ES_CONF_REG = 0x1;
+
+  // the ENTROPY_SRC interrupts can optionally be enabled
+}
+```
+
+## Noise Processing
+
+The following code snippet demonstrates initializing the ENTROPY_SRC
```

Comment:
Is this description right? This routine returns bits right, not initializes the block? It doesn't appear to use `numNoiseBits`. Also, please one sentence per line here.

Reply:
fixed.